src/loc/VoidLocButton.tsx

```diff
@@ -8,37 +8,42 @@ import DangerDialog from "../common/DangerDialog";
 import FormGroup from "../common/FormGroup";
 import Icon from "../common/Icon";
 import { useLocContext } from "./LocContext";
-import ClientExtrinsicSubmitter, { Call, CallCallback } from "src/ClientExtrinsicSubmitter";
-import { useLogionChain } from "src/logion-chain";
+import ExtrinsicSubmissionStateView from "../ExtrinsicSubmissionStateView";
+import { useLogionChain, CallCallback } from "src/logion-chain";
 
 export default function VoidLocButton() {
     const { colorTheme } = useCommonContext();
     const [ visible, setVisible ] = useState(false);
     const { mutateLocState, loc: locData } = useLocContext();
-    const [ call, setCall ] = useState<Call>();
-    const [ submissionFailed, setSubmissionFailed ] = useState<boolean>(false);
-    const { signer } = useLogionChain();
+    const { signer, submitCall, clearSubmissionState, extrinsicSubmissionState } = useLogionChain();
     const [ reason, setReason ] = useState<string>("");
 
-    const voidLocCallback = useCallback((callback: CallCallback) => {
-        return mutateLocState(async current => {
-            if(signer && (current instanceof OpenLoc || current instanceof ClosedLoc || current instanceof ClosedCollectionLoc)) {
-                return current.legalOfficer.voidLoc({
-                    reason,
-                    signer,
-                    callback,
-                });
-            } else {
-                return current;
-            }
-        });
+    const call = useMemo(() => {
+        return async (callback: CallCallback) =>
+            mutateLocState(async current => {
+                if (signer && (current instanceof OpenLoc || current instanceof ClosedLoc || current instanceof ClosedCollectionLoc)) {
+                    return current.legalOfficer.voidLoc({
+                        reason,
+                        signer,
+                        callback,
+                    });
+                } else {
+                    return current;
+                }
+            });
     }, [ signer, mutateLocState, reason ]);
 
     const clearAndClose = useCallback(() => {
-        setCall(undefined);
-        setSubmissionFailed(false);
-        setVisible(false)
-    }, [  ]);
+        clearSubmissionState();
+        setReason("");
+        setVisible(false);
+    }, [ clearSubmissionState ]);
+
+    useEffect(() => {
+        if (extrinsicSubmissionState.isSuccessful()) {
+            clearAndClose();
```

Comment:
You should call this in a callback after `await submitCall(call)`, this way you won't need that effect anymore. Note that you should wrap that statement in a `try ... catch` because in an upcoming PR, `submitCall` will throw in case of failure in call execution.

Reply:
Done as suggested.